codegen/value.go

```diff
@@ -138,8 +139,9 @@ func (v *zeroValue) Kind() parser.Kind {
 
 func (v *zeroValue) Filesystem() (Filesystem, error) {
 	return Filesystem{
-		State: llb.Scratch(),
-		Image: &solver.ImageSpec{},
+		State:    llb.Scratch(),
+		Image:    &solver.ImageSpec{},
+		Platform: specs.Platform{OS: "linux", Architecture: runtime.GOARCH},
 	}, nil
```

Comment:
So it's okay if this doesn't match the arch we're building for?

Reply:
Yeah, I have been thinking about this, not sure if this will cause problems, but it does seem wrong.  I suspect if we just pass in an empty `llb.Constraint{}` for the solves it will work when we dont have a platform defined.  I will double check this.

Reply:
So I am wrong again.  The `zeroValue` Platform is setting the platform context for the solve, effectively becoming the default platform.  If we leave out the hlb platform constraint, eventually buildkit will default to `runtime.GOOS/runtime.GOARCH`.  This will chain down to the `image "alpine"` solve, which will use the `zeroValue` Platform.  So if we leave the platform blank then we try to solve `alpine` with `darwin/amd64` which fails.  If we hardcode the platform to be `linux/runtime.GOARCH` it generally works, unless you also use `--platform linux/arm64` to change the default platform, in that case the `zeroValue` would still be the current os arch, rather than desired default platform and effectively the `--platform` argument would be ignored.

In order to fix this properly, we need to plumb the `ctx` to everywhere `ZeroValue` is called so we assign the default platform correctly.